src/main/kotlin/no/nav/syfo/behandler/BehandlerService.kt

```diff
@@ -181,18 +180,37 @@ class BehandlerService(
         } else null
     }
 
+    fun createBehandlerKontorIfMissing(
+        behandlerKontor: BehandlerKontor,
+    ) {
+        database.connection.use { connection ->
+            val pBehandlerKontor = connection.getBehandlerKontor(behandlerKontor.partnerId)
+            if (pBehandlerKontor == null) {
+                connection.createBehandlerKontor(behandlerKontor)
+            } else {
+                connection.updateBehandlerKontor(behandlerKontor, pBehandlerKontor)
+            }
+            connection.commit()
+        }
+    }
+
     fun createOrGetBehandler(
```

Comment:
Litt uheldig med en funksjon som både kan lage, hente og oppdatere en ressurs. Det er fort da man må begynne med slike flagg for å tilpasse hver enkelt bruk av funksjonen. Men den refaktoreringen bør kanskje gjøres en annen gang. Eventuelt etter denne har blitt deployet? 😊

Reply:
Ja, vi kan diskutere dette. Funksjonen brukes fra flere steder, og ble sånn etter en del forsøk på å gjøre det annerledes så vidt jeg husker. Men det kan godt være vi bør gjøre noen flere tilpasninger her.